examples/apps/ai-collab/src/components/UserPresenceGroup.tsx

```diff
@@ -5,81 +5,63 @@
-	const photoUrlsMap = new Map<string, string>();
 	const isFirstRender = useRef(true);
-	const [photoUrls, setPhotoUrls] = useState<string[]>([]);
-	const currentUserId: `id-${string}` = `id-${uuid()}`;
+	const [photoUrls, setPhotoUrls] = useState<Map<string, string>>(new Map());
 
-	/**
-	 * fetch the user's photo if it's spe client, for the tinylicious client, it will use the default photo.
-	 * */
-	const updateUserPresenceGroup = async (): Promise<void> => {
-		const clientId = process.env.NEXT_PUBLIC_SPE_CLIENT_ID;
-		const tenantId = process.env.NEXT_PUBLIC_SPE_ENTRA_TENANT_ID;
-		let photoUrl: string = "";
-
-		// spe client
-		if (tenantId !== undefined && clientId !== undefined) {
-			photoUrl = await getProfilePhoto();
-		}
-		userPresenceGroup.props.onlineUsers.local.set(currentUserId, {
-			value: { photo: photoUrl },
-		});
-		photoUrlsMap.set(currentUserId, photoUrl);
-		setPhotoUrls([...photoUrlsMap.values()]);
-
-		isFirstRender.current = false;
-	};
 
 	useEffect((): void => {
+		const userPresenceGroup = buildUserPresence(presence);
+
+		/**
+		 * fetch the user's photo if it's spe client, for the tinylicious client, it will use the default photo.
+		 * */
+		const updateUserPresenceGroup = async (): Promise<void> => {
+			const clientId = process.env.NEXT_PUBLIC_SPE_CLIENT_ID;
+			const tenantId = process.env.NEXT_PUBLIC_SPE_ENTRA_TENANT_ID;
+			let photoUrl: string = "";
+
+			// spe client
+			if (tenantId !== undefined && clientId !== undefined) {
+				photoUrl = await getProfilePhoto();
+			}
+			userPresenceGroup.props.onlineUsers.local = { photo: photoUrl };
+			setPhotoUrls((current) => { current.set(presence.getMyself().sessionId, photoUrl); return current });
+
+			isFirstRender.current = false;
+		};
+
 		if (isFirstRender.current) {
 			updateUserPresenceGroup().catch((error) => console.error(error));
 		}
 
-		userPresenceGroup.props.onlineUsers.events.on("itemUpdated", (update) => {
-			photoUrlsMap.set(update.key, update.value.value.photo);
-			setPhotoUrls([...photoUrlsMap.values()]);
+		userPresenceGroup.props.onlineUsers.events.on("updated", (update) => {
+			console.debug("Presence - 'updated' event", JSON.stringify(update));
+			setPhotoUrls((current) => { const newMap = new Map(current); newMap.set(update.client.sessionId, update.value.photo); return newMap });
 		});
-		userPresenceGroup.props.onlineUsers.events.on("itemRemoved", (update) => {
-			photoUrlsMap.delete(update.key);
-			setPhotoUrls([...photoUrlsMap.values()]);
+		presence.events.on("attendeeDisconnected", (attendee) => {
+			console.debug("Presence - 'attendeeDisconnected' event", JSON.stringify(attendee));
+			setPhotoUrls((current) => { const newMap = new Map(current); newMap.delete(attendee.sessionId); return newMap });
 		});
 	}, [
 		photoUrls,
-		photoUrlsMap,
-		userPresenceGroup.props.onlineUsers.events,
-		updateUserPresenceGroup,
 		setPhotoUrls,
+		presence,
 	]);
 
-	// Detect when the page is closed
-	useEffect(() => {
```

Comment:
Apparently unnecessary; when a tab closes, the `attendeeDisconnected` event fires correctly in other tabs with no extra code required.